File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        if not root: return True
        que = [root]
        
        def getHeight(node):
            if not node: return 0
            
            return 1+ max(getHeight(node.left), getHeight(node.right))
        
        while que:
            curNode = que.pop(0)
            left = getHeight(curNode.left)
            right = getHeight(curNode.right)
            
            if abs(left-right) > 1: 
                return False
            
            if curNode.left:
                que.append(curNode.left)
                
            if curNode.right:
                que.append(curNode.right)
                
        return True
            
        
        
#         if not root: return True
        
#         def helper(node):
#             if not node: return 0
            
#             return 1 + max(helper(node.left), helper(node.right))
        
#         return abs(helper(root.left) - helper(root.right)) < 2 and self.isBalanced(root.left) and self.isBalanced(root.right)

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    if(!root) return true;
    return Math.abs( maxHeight(root.left) - maxHeight(root.right)) < 2 && isBalanced(root.left) && isBalanced(root.right);
    
    function maxHeight(root){
        if(!root) return -1;
        return 1 + Math.max(maxHeight(root.left), maxHeight(root.right))
    }
};